**NOTEBOOK PER LA PULIZIA E IL FILTRAGGIO DEL DATASET SUI PROGETTI FINANZIATI DAL PRIN 2022**

In [36]:
import pandas as pd

In [37]:
dataset = "../dataset/prin_2022/completo_prin_2022.csv"

prin_2022 = pd.read_csv(dataset)

In [38]:
prin_2022.head()

,codice,progetto,istituzione,finanziamento
0,202224WS4K,"GREen policy shocks, Economic transition, and ...",Università degli studi di PALERMO,314509.00
1,20222X8L5C,MANAGING SUSTAINABILITY TENSIONS FOR CHANGE,Università degli studi di PADOVA,264054.00
2,20222SBH75,Entrepreneurial ecosystem and business registr...,Università Politecnica delleMARCHE,303.29
3,20222Z3CR7,Nudging under Limited Attention,Università degli Studi di PALERMO,229638.00
4,202232ZTX7,"And Justice for All: Access to Courts, Trials,...",Università Commerciale Luigi Bocconi MILANO,226.92


Bisogna correggere il formato del finanziamento: ci sono stati evidenti errori durante lo scraping (per esempio il finanziamento per il progetto 20222SBH75 è stato di circa 303mila euro e non per 303.29)

In [39]:
prin_2022.info()

<class 'pandas.DataFrame'>
RangeIndex: 1058 entries, 0 to 1057
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   codice         1058 non-null   str    
 1   progetto       1058 non-null   str    
 2   istituzione    1058 non-null   str    
 3   finanziamento  1058 non-null   float64
dtypes: float64(1), str(3)
memory usage: 33.2 KB


In [40]:
prin_2022["finanziamento"] = prin_2022["finanziamento"].astype(int)

In [41]:
prin_2022.head()

,codice,progetto,istituzione,finanziamento
0,202224WS4K,"GREen policy shocks, Economic transition, and ...",Università degli studi di PALERMO,314509
1,20222X8L5C,MANAGING SUSTAINABILITY TENSIONS FOR CHANGE,Università degli studi di PADOVA,264054
2,20222SBH75,Entrepreneurial ecosystem and business registr...,Università Politecnica delleMARCHE,303
3,20222Z3CR7,Nudging under Limited Attention,Università degli Studi di PALERMO,229638
4,202232ZTX7,"And Justice for All: Access to Courts, Trials,...",Università Commerciale Luigi Bocconi MILANO,226


In [42]:
for index, row in prin_2022.iterrows():
    if len(str(int(row["finanziamento"]))) == 3:
        prin_2022.at[index, "finanziamento"] = row["finanziamento"] * 1000

In [43]:
finanziamento_medio = prin_2022["finanziamento"].mean()
int(finanziamento_medio)

252053

In [44]:
prin_2022.head()

,codice,progetto,istituzione,finanziamento
0,202224WS4K,"GREen policy shocks, Economic transition, and ...",Università degli studi di PALERMO,314509
1,20222X8L5C,MANAGING SUSTAINABILITY TENSIONS FOR CHANGE,Università degli studi di PADOVA,264054
2,20222SBH75,Entrepreneurial ecosystem and business registr...,Università Politecnica delleMARCHE,303000
3,20222Z3CR7,Nudging under Limited Attention,Università degli Studi di PALERMO,229638
4,202232ZTX7,"And Justice for All: Access to Courts, Trials,...",Università Commerciale Luigi Bocconi MILANO,226000


Il finanziamento **totale** ai progetti PRIN 2022 nei settori SH 1 - 7 è stato di circa **266 milioni di euro**

In [45]:
finanziamenti_totali = prin_2022["finanziamento"].sum()

int(finanziamenti_totali)

266672767

Il finanziamento **medio** ai progetti PRIN 2022 nei settori SH 1 - 7 è stato di circa **252 mila euro**

In [46]:
finanziamento_medio = prin_2022["finanziamento"].mean()
int(finanziamento_medio)

252053

Proviamo a filtrare i progetti nel campo delle DH

In [47]:
parole_chiave_dh = [
    #digital x
    "digital humanities", "digital archive", "digital preservation", "digital heritage",
    "digital transition", "digital mapping", "digital catalogue", "digital repertoire",
    "digital atlas", "digitisation", "metadata", "digital repository",
    "digital enhancement", "archival infrastructures", "digital curation", "digital restoration",
    "digital linguistics", "digital linguistic", "digital cartography",
    "digital reconstruction",  "virtual heritage",
    "computer vision", "digital broadcasting archives", "3D modeling",
    "digital dissemination"

    # LINGUISTICA COMPUTAZIONALE E ANALISI TESTUALE
    "computational linguistics", "computational linguistic",
    "textual", "corpus", "semantic", "syntax", "lexical", "philology", 
    "scholarly edition", "encoding", "paratext", "data-driven", "data oriented", 
    "computational", "algorithmic", "machine learning", "human-machine interaction",
    "distributional semantic analysis", "linked data", "interlinking", "text mining",
    "named entity recognition", "stylometry", "digital papyrology", "scholarly digital edition",
    "digital epigraphy", "formal models", "conceptual models",

    # --- GEOGRAFIA DIGITALE, CARTOGRAFIA E ARCHEOLOGIA ---
    "GIS", "preventive archaeology", "spatial analysis", "deep mapping",
    "historical GIS",

    "knowledge graph", "digital journalism", "data journalism",
]


parole_chiave_sh = [
    # Economia e Finanza
    "economics", "gdp", "inflation", "unemployment", "taxation", "bank", "banking", "economy",

    # Sociologia, Antropologia e Demografia
    "sociology", "demography", "migration", "inequality", "gender", 
    "ethnicity", "anthropological", "ritual", "voting",

    # Scienze Politiche, Istituzioni e Relazioni Internazionali
    "politics", "political", "democracy", "diplomacy", "institution", 
    "legislation", "regulation", "treaty",

    # Psicologia e Scienze Cognitive
    "psychology", "psychological", "cognition", "perception", 
    "anxiety", "depression", "personality",

    # Geografia Umana e Media
    "urbanization", "pedagogy"
]

In [48]:
indici_progetti_dh = []

for index, row in prin_2022.iterrows():
    progetto = row["progetto"].lower()
    
    is_dh = any(chiave_dh in progetto for chiave_dh in parole_chiave_dh)
    is_sh = any(chiave_sh in progetto for chiave_sh in parole_chiave_sh)

    if is_dh and not is_sh:
        indici_progetti_dh.append(index)

In [49]:
df_progetti_dh = prin_2022.loc[indici_progetti_dh].reset_index(drop=True)

df_progetti_dh

,codice,progetto,istituzione,finanziamento
0,2022KFLF3E,"EquAl: Equitable Algorithms, Promoting Fairnes...",Università degli studi di BOLOGNA,242691
1,20228W9NE7,Caring for Children with an Immigrant Backgrou...,Università degli studi di ROMA La Sapienza,220000
2,202297CKET,ALGOLIT: Understanding and boosting algorithmi...,Università degli Studi di TRENTO,197997
3,2022YRJ83A,Feedback culture: assessing the effects of alg...,Università degli Studi di MILANO,211378
4,2022Z78M8J,The Digital Transition and the World of Work: ...,Università degli studi di BARI ALDO MORO,246248
5,202233C93X,MetaLing Corpus: Creating a corpus of English ...,Università degli Studi di MILANO,106675
6,202249CYPP,A.L.AR..CO (Archivio Linguistico ARbëresh e lt...,Università della CALABRIA,325376
7,20225MH5NM,Disentangling heteRogeneity in autism: from EA...,Università degli Studi di TRIESTE,295202
8,20225PJ2JM,Interactions between visuo-spatial attention a...,Università degli studi di ROMA La Sapienza,248000
9,2022A74X79,Corpus SIM (Senectus Ipsa Morbus) - Spontaneou...,Università degli studi di Napoli Federico II,255106


In [50]:
print(f"Trovati {len(df_progetti_dh)} progetti di Digital Humanities:")

for index, row in df_progetti_dh.iterrows():
    print(f"{index+1} -> {row["progetto"]}")


Trovati 51 progetti di Digital Humanities:
1 -> EquAl: Equitable Algorithms, Promoting Fairness and Countering AlgorithmicDiscrimination Through Norms and Technologies.
2 -> Caring for Children with an Immigrant Background: A Cross-Sequential Longitudinal Study on Kindness Pillars and the Role of Dispositional and Contextual Factors inEarly Childhood
3 -> ALGOLIT: Understanding and boosting algorithmic literacy
4 -> Feedback culture: assessing the effects of algorithmic recommendations onplatformized consumption - ALGOFEED
5 -> The Digital Transition and the World of Work: Labour markets, Organizations, Jobquality and Industrial Relations
6 -> MetaLing Corpus: Creating a corpus of English linguistics metalanguage from the16th to the 18th century
7 -> A.L.AR..CO (Archivio Linguistico ARbëresh e ltalo-romanzo in COntatto)Testo italianoA.L.AR.l.Co (Arberesh and Italo-Romance language corpus in contact)
8 -> Disentangling heteRogeneity in autism: from EArly behavioral markers toindividual 

In [51]:
finanziamenti_progetti_DH = df_progetti_dh["finanziamento"].sum()
int(finanziamenti_progetti_DH)

12419545